src/Aspire.Hosting.Azure.Sql/AzureSqlServerResource.cs

```diff
@@ -75,13 +75,21 @@ public ReferenceExpression ConnectionStringExpression
     public override ResourceAnnotationCollection Annotations => InnerResource?.Annotations ?? base.Annotations;
 
     /// <summary>
-    /// A dictionary where the key is the resource name and the value is the database name.
+    /// A dictionary where the key is the resource name and the value is the Azure SQL database resource.
     /// </summary>
-    public IReadOnlyDictionary<string, string> Databases => _databases;
+    public IReadOnlyDictionary<string, AzureSqlDatabaseResource> AzureSqlDatabases => _databases;
 
-    internal void AddDatabase(string name, string databaseName)
+    /// <summary>
+    /// A dictionary where the key is the resource name and the value is the Azure SQL database name.
+    /// </summary>
+    public IReadOnlyDictionary<string, string> Databases => _databases.ToDictionary(
```

Comment:
@eerhardt should this be obsolete too? Or just track that this should be removed in 10.0?

Reply:
We shouldn't remove APIs without obsoleting them for a release first.

If we don't think we want this API anymore, we can obsolete it. I'd be fine with that.

Reply:
> We shouldn't remove APIs without obsoleting them for a release first.

Really? Meaning we won't remove anything in 10.0 (or at least try) if we haven't obsoleted it in 9.3 or anything before 10?

Reply:
That's my opinion, yes. I'm sure there will be cases where we feel we have to make a API breaking change without obsoleting first. But I think we should try to obsolete for a release first, before removing APIs.

Thoughts @DamianEdwards @davidfowl @mitchdenny ?

Reply:
Yep